particle_grid/unit_test/tstPeriodic.hpp

```diff
@@ -0,0 +1,183 @@
+/****************************************************************************
+ * Copyright (c) 2018-2020 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Periodic.hpp>
+
+#include <Cabana_Core.hpp>
+#include <Cajita.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <gtest/gtest.h>
+
+#include <mpi.h>
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+namespace Test
+{
+
+//---------------------------------------------------------------------------//
+void testMigrate( int test_type )
+{
+    // Create the MPI partitions.
+    Cajita::UniformDimPartitioner partitioner;
+    auto ranks_per_dim = partitioner.ranksPerDimension( MPI_COMM_WORLD, {} );
+
+    // Create the global MPI decomposition mesh.
+    std::array<double, 3> low_corner = {-3.2, -0.5, 2.2};
+    std::array<double, 3> high_corner = {-0.2, 5.9, 4.2};
+    auto global_mesh = Cajita::createUniformGlobalMesh( low_corner, high_corner,
+                                                        ranks_per_dim );
+
+    // Create the global grid.
+    std::array<bool, 3> is_periodic = {true, true, true};
+    auto global_grid = Cajita::createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                                 is_periodic, partitioner );
+
+    // Create a  grid local_grid.
+    auto local_grid = Cajita::createLocalGrid( global_grid, 1 );
+    auto local_mesh = Cajita::createLocalMesh<TEST_DEVICE>( *local_grid );
```

Comment:
Does it seem reasonable to have to make all of these objects for cases like MD with short-range only where you don't need a mesh? I could see automating how often `migrate` is called based on setting some halo in MD. That would then eliminate the need to tune how often migrate happens (e.g. every 100 time steps or something). It would instead be based on some distance criterion which would correspond to a halo width. In that case you would want to create all of these objects then as your cell size would have some distance relative to something meaningful in the simulation (like the interaction distance).

Reply:
Note this extra grid we would use to choose communications would also be the same grid from which we would build `LinkedCellList` objects.

Reply:
And now that I think about it we need a `LinkedCellList` that uses Cajita instead of the grid we have as an implementation detail. Even more reason for all of this stuff to just be in `core`.

Reply:
I guess I'm not sure how to get around creating all these objects, but I have considered specializations of the mesh/grid that don't hold as much information since we basically only need the bounds of the box in MD

And since we have to rebuild the neighbor list every so often, we do the same with migration, but you're right that it could potentially be done automatically with a check for anything crossing halo boundaries. 

I have the `LinkedCellList` on my list after the `PeriodicHalo`